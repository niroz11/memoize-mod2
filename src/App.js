import React, { Component } from "react";
import Scopecomp from "./Scopecomp.js";

class App extends Component {
  constructor() {
    super();
    this.state = {
      data: [],
      score: 0,
      clickValue: "",
      question: -1,
      showCard: true
    };
  }
  componentDidMount() {
    fetch("http://memoize-datasets.herokuapp.com/api/v1/nirajData")
      .then(response => response.json())
      .then(data => {
        this.setState({
          data: data.nirajData
        });
      })
      .catch(error => {
        throw new Error(error);
      });
  }

  handleClick = event => {
    let question = this.state.question;
    question++;
    this.setState({
      clickValue: event.target.name,
      question
    });
  };

  incrementQuestion = () => {
    let newQuestion = this.state.question;
    newQuestion++;

    this.setState({
      question: newQuestion
    });
  };
  
  getMainMenu = () => {
    this.setState({
      clickValue: "",
      question: -1
    });
  };

  updateScore = () => {
    this.setState({
      score: this.state.score + 10
    });
  };
  toggleCard = () => {
    this.setState({
      showCard: !this.state.showCard
    });
  };

  render() {
    const incorrect = JSON.parse(localStorage.getItem("wrong")) || [];

    const { clickValue, data } = this.state;

    const buttons = () => {
      return (
        <div className="buttons">
          <button
            className="quiz-button"
            onClick={this.handleClick}
            name="quiz"
          >
            Take the quiz
          </button>
          <button
            className="redo-button"
            onClick={this.handleClick}
            name="redo"
          >
            Try again
          </button>
        </div>
      );
    };
    return (
      <div className="entire-page">
        <div className="welcome-page">
          <p className="title">JS Fun</p>

          <p className="score">{this.state.score}</p>

          {clickValue === "" && buttons()}
          <div className="question-section">
            <Scopecomp
              data={clickValue === "quiz" ? data : incorrect}
              question={this.state.question}
              increment={this.incrementQuestion}
              score={this.state.score}
              updateScore={this.updateScore}
              mainMenu={this.getMainMenu}
              localStorage={this.saveToLocalStorage}
              toggleCard={this.toggleCard}
              showCard={this.state.showCard}
            />
          </div>
        </div>
      </div>
    );
  }
}

export default App;
