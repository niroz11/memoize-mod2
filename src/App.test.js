import React from "react";
import App from "./App";
import { shallow } from "enzyme";

describe("App", () => {
  

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <App />
    );
  });
	
  it("should match the snapshot with all data passed in", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should have a proper default state", () => {
    expect(wrapper.state()).toEqual({ data: [], score: 0,clickValue: "",question: -1,showCard: true  });
  });
  it("should increase question count", () => {
    expect(wrapper.state()).toEqual({ data: [], score: 0,clickValue: "",question: -1,showCard: true  });
    wrapper.find(".quiz-button").simulate("click", { target: { clickValue: "scope", question: 0}});
    expect(wrapper.state("question")).toEqual(0);
  });
  it("should increase question count", () => {
    expect(wrapper.state()).toEqual({ data: [], score: 0,clickValue: "",question: -1,showCard: true  });
    wrapper.find(".redo-button").simulate("click", { target: { clickValue: "scope", question: 0}});
    expect(wrapper.state("question")).toEqual(0);
  });
  it("Increment question should change state when invoked", () => {
    expect(wrapper.instance().incrementQuestion());
  })
  it("It should change state when getMainMenu is invoked", () => {
    expect(wrapper.instance().getMainMenu());
  })
  it("It should change state when updateScore is invoked", () => {
    expect(wrapper.instance().updateScore());
  })
  it("It should change state when toggleCard is invoked", () => {
    expect(wrapper.instance().toggleCard());
  })

  
});

