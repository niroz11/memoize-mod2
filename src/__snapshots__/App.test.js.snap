// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in 1`] = `
<div
  className="entire-page"
>
  <div
    className="welcome-page"
  >
    <h1
      className="hello"
    >
      Memoize
    </h1>
    <p
      className="score"
    >
      0
    </p>
    <div
      className="buttons"
    >
      <button
        className="quiz-button"
        name="quiz"
        onClick={[Function]}
      >
        Take the quiz
      </button>
      <button
        className="redo-button"
        name="redo"
        onClick={[Function]}
      >
        Question Missed
      </button>
    </div>
    <div
      className="question-section"
    >
      <Scopecomp
        data={Array []}
        increment={[Function]}
        mainMenu={[Function]}
        question={-1}
        score={0}
        showCard={true}
        toggleCard={[Function]}
        updateScore={[Function]}
      />
    </div>
  </div>
</div>
`;
